### PR TITLE
Resolve props assets on client

### PR DESCRIPTION
**Pull request by TrySound**
Before we checked looked for all asset ids in props, loaded assets and replaced ids with resolved assets. This is all on server. We faced a few problems with it.

- requires to maintain two versions of types: stored in db and resolved for client
- easy to break project by writing to db wrong version

Here I provide to sdk both props and assets and resolve on client with almost same amount of code.

Also replaced store setter with context, see https://github.com/webstudio-is/webstudio-designer/issues/213

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - test it on preview
- [ ] hi @istarkov, I need you to do
  - detailed review (read every line)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/canvas/canvas.tsx [hunk 1]
```diff
@@ -73,12 +79,21 @@ const useElementsTree = () => {
...
     }
 
-    return createElementsTree({
-      sandbox: true,
-      instance: rootInstance,
-      Component: WrapperComponentDev,
-      onChangeChildren,
-    });
+    return (
+      <ReactSdkProvider
```
Comment (kof) [inline apps/designer/app/canvas/canvas.tsx:83]: This makes canvas fully coupled to react SDK ... how is this supposed to work some day with a different framework?

Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx:83]: It is coupled anyway. With other framework we'll use another way to inject stores.

Comment (kof) [reply apps/designer/app/canvas/canvas.tsx:83]: how is it coupled anyway? we are trying to remove every coupling in every step we do ...
I think react provider that has to be provided is not the ideal way of doing it, SDK should provide something agnostic so we don't have to assume on canvas that its already using react.

there is a bunch of options

Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx:83]: Renderer is full react. It won't be reused in any way. We will reimplement it in other frameworks.

Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx:83]: I can put provider inside createElementsTree if you want

Comment (kof) [reply apps/designer/app/canvas/canvas.tsx:83]: Anything works for me that doesn't create more coupling with react in SDK. Currently you require SDK to provide this provider

Comment (TrySound) [reply apps/designer/app/canvas/canvas.tsx]: done

**Review by kof — APPROVED** [review]
I don't like how we do SDK stuff right now in general, still way too much is put in there and creating a separate SDK in a new framework is currently practically impossible 🤷‍♂️

Just food for thoughts

**Comment by TrySound** [comment]
Almost everything in there is renderer and components. Isn't it what supposed to be there?

**Comment by kof** [comment]
Yeah, the problem is more like how do we expect someone writing an SDK for a different framework e.g. vue. There is quite a bit of stuff they have to implement that has to do with the designer. Ideal situation is that they just provide components and are done.

File: packages/react-sdk/src/props.ts [hunk 2]
```diff
@@ -50,20 +42,24 @@ export const useInstanceProps = (instanceId: Instance["id"]) => {
 // this utility is be used for image component in both designer and preview
 // so need to optimize rerenders with computed
 export const usePropAsset = (instanceId: Instance["id"], name: string) => {
+  const { propsByInstanceIdStore, assetsStore } = useContext(ReactSdkContext);
   const assetStore = useMemo(() => {
-    return computed(propsByInstanceIdStore, (propsByInstanceId) => {
-      const instanceProps = propsByInstanceId.get(instanceId);
-      let asset: undefined | Asset = undefined;
-      if (instanceProps) {
+    return computed(
```
Comment (istarkov) [inline packages/react-sdk/src/props.ts:47]: I may be wrong, but if we create a computed atom inside the hook, it subscribes to other atoms holding references. 

Any chance that this unmount is called?
https://github.com/nanostores/nanostores/blob/main/computed/index.js#L26

**Review by istarkov — APPROVED** [review]
Except I have feeling that creating computed inside hook can cause memory leak, everything else is ok

**Pull request merged**